subsys/settings/src/settings.c

```diff
@@ -17,25 +17,40 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(settings, CONFIG_SETTINGS_LOG_LEVEL);
 
+#if defined(CONFIG_SETTINGS_RAM_HANDLERS)
 sys_slist_t settings_handlers;
+#endif /* CONFIG_SETTINGS_RAM_HANDLERS */
+
 struct k_mutex settings_lock;
 
 
 void settings_store_init(void);
 
 void settings_init(void)
 {
+#if defined(CONFIG_SETTINGS_RAM_HANDLERS)
 	sys_slist_init(&settings_handlers);
+#endif /* CONFIG_SETTINGS_RAM_HANDLERS */
 	settings_store_init();
 }
 
+#if defined(CONFIG_SETTINGS_RAM_HANDLERS)
 int settings_register(struct settings_handler *handler)
 {
 	int rc;
 	struct settings_handler *ch;
 
 	k_mutex_lock(&settings_lock, K_FOREVER);
 
+#if defined(CONFIG_SETTINGS_ROM_HANDLERS)
```

Comment:
You should probably document that static defined handlers takes precedence over dynamic ones, though this eliminates the possibility of overriding system handlers, either way Im fine we just have to clearly state what are their priority somewhere.

Reply:
Well, it is the other way around, dynamic handlers take precedence over static ones, if it finds a match after the ROM handlers in the RAM handlers it will take the RAM handlers. I had the idea of overriding handlers in mind when writing.

Reply:
The overriding is something that can be used by mistake though so perhaps it is fine to not allow it at runtime, for the static one I guess we cannot detect duplicate handlers though expect perhaps if we declare some variable with the handler prefix to cause build time errors but that will increase the code size so I don't think it is worth it.